webserver/src/main/java/io/kestra/webserver/controllers/ExecutionController.java

```diff
@@ -962,4 +964,58 @@ public Flowable<Event<Execution>> follow(
                 }
             });
     }
+
+    @ExecuteOn(TaskExecutors.IO)
+    @Get(uri = "executions/{executionId}/file/preview", produces = MediaType.TEXT_JSON)
+    @Operation(tags = {"Executions"}, summary = "Get file preview for an execution")
+    public HttpResponse<?> filePreview(
+        @Parameter(description = "The execution id") @PathVariable String executionId,
+        @Parameter(description = "The internal storage uri") @QueryValue URI path
+    ) throws IOException {
+        HttpResponse<StreamedFile> httpResponse = this.validateFile(executionId, path, "/api/v1/executions/{executionId}/file?path=" + path);
+        if (httpResponse != null) {
+            return httpResponse;
+        }
+
+        InputStream fileStream = storageInterface.get(path);
+
+        String extension = FilenameUtils.getExtension(path.toString());
+
+        if (ImageFileExtension.isImageFileExtension(extension)) {
+            byte[] imageBytes = IOUtils.toByteArray(fileStream);
+            String base64Image = Base64.getEncoder().encodeToString(imageBytes);
+
+            return HttpResponse.ok(PreviewResponse.builder().extension(extension).content(base64Image).build());
+        }
+        BufferedReader reader = new BufferedReader(new InputStreamReader(fileStream));
+        String line = reader.readLine();
+        int lineCount = 0;
+
+        StringBuilder contentBuilder = new StringBuilder();
+
+        while (line != null && lineCount < 100) {
+            contentBuilder.append(line);
+            lineCount++;
+            if ((line = reader.readLine()) != null) {
+                contentBuilder.append("\n");
+            }
+        }
+
+        return HttpResponse.ok(PreviewResponse.builder().extension(extension).content(truncateStringSize(contentBuilder.toString())).build());
```

Comment:
Truncating the file after it has been read is not a good idea as the file will still be loaded entirely in memory.

Reply:
agree also